### PR TITLE
TL-182 Deployable shield

### PR DESCRIPTION
**Pull request by Lumipharon**
## About The Pull Request
Adds the TL-182 deployable shield.

image

It's a shield that provides inferior protection compared to a normal shield, but you can deploy it on the ground, turning it into a barricade.
This behaves more or less like a normal cade (you can wire and repair it etc, the wiring staying when you undeploy it). It has some soft armour but can't be upgraded like a normal cade, so sits somewhere between a metal and plasteel cade in terms of durability.
Its coverage is 100 instead of 128, which means it will still block all projectiles, but projectiles with accuracy over 100 will have an easier time bypassing it.
As a shield, it will block less damage, has lower armour and HP than a normal shield, but has slightly lower slowdown.

It is a normal size item so fits in bags, but can be worn on your back for TMNT larp.

Added to Combat Patrol in a few basic unga and leader loadouts that were a bit weaker offensively, to give them more defensive utility.

Added to HvX in req for 3 points and 6 into the vendors. I dunno if they'll actually be worth using, but doesn't hurt to have them there.

## Why It's Good For The Game
A new mobile defensive item
Deployables r kool

## Changelog
:cl:
add: Adds a deployable shield for marines, available in req and normal vendors, and in certain Combat Patrol loadouts
/:cl:

### CODE REVIEW

**Comment by FosterGrime-Heymynameslouie** [comment]
this sounds alot like what deployable barricade supposed to be

simmilar to CM deployable cades

spedde merhg

**Comment by maikilangiolo** [comment]
Add it to regular rounds, will make using fixed guns tolerable on the front

**Comment by FosterGrime-Heymynameslouie** [comment]
wait so what happens if it fully gets destroyed? will it have a destroyed state where everything can pass and you can still pick it up and repair it like a normal shield or will it fully get destroyed ie no longer recoverable

**Comment by Lumipharon** [comment]
same as a shield or cade, fully destroyed = gone.

File: code/datums/elements/deployable_item.dm [hunk 1]
```diff
@@ -141,7 +141,7 @@
 	attached_item.max_integrity = deployed_machine.max_integrity
 	attached_item.obj_integrity = deployed_machine.obj_integrity
 
-	deployed_machine.internal_item = null
+	deployed_machine:internal_item = null
```
Comment (TiviPlus) [inline code/datums/elements/deployable_item.dm:144]: Absolutely not

File: code/game/objects/structures/barricade.dm [hunk 2]
```diff
@@ -1160,3 +1160,130 @@
...
+		return
+	var/mob/living/carbon/human/user = usr
+	if(over_object != user || !in_range(src, user))
+		return
+	if(CHECK_BITFIELD(internal_item.flags_item, DEPLOYED_WRENCH_DISASSEMBLE))
+		to_chat(user, "<span class = 'notice'>You cannot disassemble [src] without a wrench.</span>")
+		return
+	disassemble(user)
+
+obj/structure/barricade/deployable/wire()
```
Comment (TiviPlus) [inline code/game/objects/structures/barricade.dm:1212]: missing /
```suggestion

/obj/structure/barricade/deployable/wire()
```

File: code/datums/elements/deployable_item.dm [hunk 3]
```diff
@@ -117,8 +117,8 @@
 
 //Handles the conversion of Machine into Item. 'source' should be the Machine. User is the one undeploying. It can be undeployed without a user, if so, the var 'location' is required. If 'source' is not /obj/machinery/deployable then 'undeploying' should be the item to be undeployed from the machine.
 /datum/element/deployable_item/proc/finish_undeploy(datum/source, mob/user)
-	var/obj/machinery/deployable/deployed_machine = source //The machinethat is undeploying should be the the one sending the Signal
-	var/obj/item/attached_item  = deployed_machine.internal_item //Item the machine is undeploying
+	var/obj/deployed_machine = source //The machinethat is undeploying should be the the one sending the Signal
+	var/obj/item/attached_item  = deployed_machine:internal_item //Item the machine is undeploying
```
Comment (TiviPlus) [inline code/datums/elements/deployable_item.dm:121]: absolutely not

**Review by TiviPlus — CHANGES_REQUESTED** [review]

File: code/datums/elements/deployable_item.dm (hunk 3 above)
Comment (TiviPlus) [reply code/datums/elements/deployable_item.dm:121]: Ive thought about it and putting it on obj is probably teh best idea but you should probably ask brave and spooky for second opinions

**Comment by Autisem** [comment]
only want to ask if letting it both fit in backpack AND being used as a shield is good idea. considering the normal shield cannot fit in backpack. also yeah being able to deploy on it would be very cool

**Comment by Derrick2Electricboogaloo** [comment]
I thought we already had one.

File: code/game/objects/items/weapons/shields.dm [hunk 4]
```diff
@@ -183,6 +183,36 @@
...
+	icon_state = "folding_shield"
+	flags_equip_slot = ITEM_SLOT_BACK
+	w_class = WEIGHT_CLASS_NORMAL
+	max_integrity = 300
+	integrity_failure = 50
+	soft_armor = list(MELEE = 35, BULLET = 30, LASER = 20, ENERGY = 40, BOMB = 25, BIO = 50, "rad" = 0, FIRE = 0, ACID = 30)
+	slowdown = 0.3
+	flags_item = IS_DEPLOYABLE
+	///The item this deploys into
+	var/deployable_item = /obj/structure/barricade/deployable
```
Comment (BraveMole) [inline code/game/objects/items/weapons/shields.dm:200]: Why is this a new var ?

File: code/game/objects/objs.dm [hunk 5]
```diff
@@ -40,6 +40,9 @@
 	///odds of a projectile hitting the object, if throwpass is true and the object is dense
 	var/coverage = 50
 
+	///Internal deployable item
+	var/obj/item/internal_item
```
Comment (BraveMole) [inline code/game/objects/objs.dm:44]: @TiviPlus how about a get internal item proc ?

**Review by BraveMole — CHANGES_REQUESTED** [review]

File: code/game/objects/objs.dm (hunk 5 above)
Comment (TiviPlus) [reply code/game/objects/objs.dm:44]: Oh true that is better do this instead

File: code/game/objects/structures/barricade.dm [hunk 6]
```diff
@@ -1160,3 +1160,130 @@
...
+/obj/structure/barricade/deployable/Destroy()
+	if(internal_shield)
+		QDEL_NULL(internal_shield)
+	return ..()
+
+/obj/structure/barricade/deployable/MouseDrop(over_object, src_location, over_location)
+	if(!ishuman(usr))
+		return
+	var/mob/living/carbon/human/user = usr
+	if(over_object != user || !in_range(src, user))
```
Comment (TiviPlus) [inline code/game/objects/structures/barricade.dm:1205]: missing adjacency/incapacitated checks and so

File: code/game/objects/structures/barricade.dm [hunk 7]
```diff
@@ -1160,3 +1160,130 @@
...
+	. = ..()
+	//makes the shield item wired as well
+	internal_shield.is_wired = TRUE
+	internal_shield.modify_max_integrity(max_integrity + 50)
+
+//repairable but not upgradable
+/obj/structure/barricade/deployable/attackby(obj/item/I, mob/user, params)
+	. = ..()
+
+	if(istype(I, /obj/item/stack/sheet/metal))
```
Comment (TiviPlus) [inline code/game/objects/structures/barricade.dm:1222]: why is there so much copy paste for this?

Comment (TiviPlus) [reply code/game/objects/structures/barricade.dm:1222]: also meaning the welder act proc

**Review by TiviPlus — CHANGES_REQUESTED** [review]

File: code/game/objects/objs.dm (hunk 5 above)
Comment (Lumipharon) [reply code/game/objects/objs.dm]: Done

File: code/game/objects/structures/barricade.dm (hunk 6 above)
Comment (Lumipharon) [reply code/game/objects/structures/barricade.dm]: Done, although this was the same as other deployable, so not sure if it was actually an issue

File: code/game/objects/structures/barricade.dm (hunk 7 above)
Comment (Lumipharon) [reply code/game/objects/structures/barricade.dm]: Not sure why I didn't make it a child of metal cades originally, it is now.

File: code/game/objects/items/weapons/shields.dm [hunk 8]
```diff
@@ -183,6 +183,36 @@
...
+	///Time to deploy
+	var/deploy_time = 1 SECONDS
+	///Time to undeploy
+	var/undeploy_time = 1 SECONDS
+	///Whether it is wired
+	var/is_wired = FALSE
+
+/obj/item/weapon/shield/riot/marine/deployable/Initialize()
+	. = ..()
+	if(deployable_item)
```
Comment (BraveMole) [inline code/game/objects/items/weapons/shields.dm:210]: Why would deployable shit not have a deployable item

**Review by BraveMole — APPROVED** [review]
Good pr

File: code/game/objects/items/weapons/shields.dm (hunk 8 above)
Comment (Lumipharon) [reply code/game/objects/items/weapons/shields.dm:210]: Yeah they shouldn't unless someone fucks up I guess. I saw similar checks used for this element elsewhere though, but do you think I should just remove it?

**Comment by lKiseki** [comment]
Testing it live, the shields do not seem damageable by most sources of damage, with the exception of xeno slashes.
I tried attacking a deployed shield with the following:
- Marine with a spear. No damage.
- Marine with an AR-12. No damage.
- Marine with a HMG-08. No damage.
- SoM with a spear. No damage.
- SoM with a lasgun. No damage.
- Xeno acid spit, no damage.

**Comment by comfyorange** [comment]
#### Test Merge Removed

**Comment by Lumipharon** [comment]
> Testing it live, the shields do not seem damageable by most sources of damage, with the exception of xeno slashes. I tried attacking a deployed shield with the following:
> 
>     * Marine with a spear. No damage.
> 
>     * Marine with an AR-12. No damage.
> 
>     * Marine with a HMG-08. No damage.
> 
>     * SoM with a spear. No damage.
> 
>     * SoM with a lasgun. No damage.
> 
>     * Xeno acid spit, no damage.

Removal of rad armour was causing it to runtime and not apply damage. Thus proving that removing rad armour was a mistake.
Also fixed.

**Pull request merged**